pkg/monitortests/testframework/operatorloganalyzer/operator_lease_check.go

```diff
@@ -0,0 +1,143 @@
+package operatorloganalyzer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/monitortestframework"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+	"k8s.io/client-go/rest"
+)
+
+type operatorLeaseCheck struct {
+}
+
+func OperatorLeaseCheck() monitortestframework.MonitorTest {
+	return &operatorLeaseCheck{}
+}
+
+func (w *operatorLeaseCheck) StartCollection(ctx context.Context, adminRESTConfig *rest.Config, recorder monitorapi.RecorderWriter) error {
+	return nil
+}
+
+func (w *operatorLeaseCheck) CollectData(ctx context.Context, storageDir string, beginning, end time.Time) (monitorapi.Intervals, []*junitapi.JUnitTestCase, error) {
+	return nil, nil, nil
+}
+
+func (*operatorLeaseCheck) ConstructComputedIntervals(ctx context.Context, startingIntervals monitorapi.Intervals, recordedResources monitorapi.ResourcesMap, beginning, end time.Time) (monitorapi.Intervals, error) {
+	ret := monitorapi.Intervals{}
+
+	leaseIntervals := startingIntervals.Filter(func(eventInterval monitorapi.Interval) bool {
+		if eventInterval.Message.Reason == monitorapi.LeaseAcquired || eventInterval.Message.Reason == monitorapi.LeaseAcquiringStarted {
+			return true
+		}
+		return false
+	})
+
+	podToLeaseIntervals := map[string][]monitorapi.Interval{}
+
+	for _, interval := range leaseIntervals {
+		podToLeaseIntervals[interval.Locator.OldLocator()] = append(podToLeaseIntervals[interval.Locator.OldLocator()], interval)
+	}
+
+	errs := []error{}
+	for podLocator, perPodLeaseIntervals := range podToLeaseIntervals {
+		var lastAcquiringFrom *time.Time
+		for _, interval := range perPodLeaseIntervals {
+			switch interval.Message.Reason {
+			case monitorapi.LeaseAcquiringStarted:
+				// only overwrite if there isn't one already
+				if lastAcquiringFrom == nil {
+					lastAcquiringFrom = &interval.From
+				}
+
+			case monitorapi.LeaseAcquired:
+				if lastAcquiringFrom == nil {
+					errs = append(errs, fmt.Errorf("missing acquiring stage for %v", podLocator))
```

Comment:
are those errors going to fail the test?

Reply:
> are those errors going to fail the test?

These will, yes. But they also mean that somehow the beginning of the log was missing.